packages/pigeon/platform_tests/shared_test_plugin_code/lib/integration_tests.dart

```diff
@@ -123,6 +123,43 @@ void runPigeonIntegrationTests(TargetGenerator targetGenerator) {
           await api.createNestedString(sentString);
       expect(receivedObject.values.aString, sentString);
     });
+
+    testWidgets(
+        'Arguments of multiple types serialize and deserialize correctly',
+        (WidgetTester _) async {
+      final HostIntegrationCoreApi api = HostIntegrationCoreApi();
+      const String aString = 'this is aString';
+      const bool aBool = false;
+      const int anInt = 42;
+
+      final AllTypes echoObject =
+          await api.sendMultipleTypes(aBool, anInt, aString);
+      expect(echoObject.anInt, anInt);
+      expect(echoObject.aBool, aBool);
+      expect(echoObject.aString, aString);
+    });
+
+    testWidgets('Ints serialize and deserialize correctly',
+        (WidgetTester _) async {
+      final HostIntegrationCoreApi api = HostIntegrationCoreApi();
+
+      const int inInt = -13;
+      final int? anInt = await api.echoInt(inInt);
+      expect(anInt, inInt);
+    });
+
+    testWidgets('booleans serialize and deserialize correctly',
+        (WidgetTester _) async {
+      final HostIntegrationCoreApi api = HostIntegrationCoreApi();
+
+      bool sentBool = true;
+      do {
+        final bool? receivedBool = await api.echoBool(sentBool);
+        expect(receivedBool, sentBool);
+
+        sentBool = !sentBool;
+      } while (!sentBool);
```

Comment:
This took me a minute to parse; you could loop over an array of bool values instead to make it more obvious what's being done here:

```
for (final bool sentBool in <bool>[true, false]) {
  final bool? receivedBool = await api.echoBool(sentBool);
  expect(receivedBool, sentBool);
}
```

Reply:
Well that's no fun! I did this as a weird little idea in my head and forgot to simplify it later.